File: lab2/FileReverse.java
// ----------------------------------------------------------------------
// Duy Nguyen
// W1475318
// CS 12M lab2
// 10/04/16
// FileReverse.java
// Takes two command line arguments, reading lines of input and printing
// each token backwards to the output file on a line by itself
// -----------------------------------------------------------------------
import java.io.*;
import java.util.Scanner;

class FileReverse{
    public static void main(String[] args) throws IOException{

	int lineNumber = 0;

	// check number of command line arguments is at least 2
        if(args.length < 2){
	    System.out.println("Usage: FileCopy <input file> <output file>");
	    System.exit(1);
        }

	// open files
	Scanner in = new Scanner(new File(args[0]));
	PrintWriter out = new PrintWriter(new FileWriter(args[1]));

	// read lines from in, extract and pring tokens from each line
	while(in.hasNextLine()){
	    lineNumber++;

	    // trim leading and trailing spaces, then add one trailing space so
	    // split works on blank lines
	    String line = in.nextLine().trim() + " ";

	    // split line around white space
	    String[] token = line.split("\\s+");

	    // print out tokens
	    int n = token.length;
	    for(int i = 0; i < n; i++){
	        out.println(stringReverse(token[i], token[i].length()));
	    }
	
        }

        // close files
        in.close();
	out.close();

    }
    public static String stringReverse(String s, int n){
        if(s.length() == 1){
	    return s;
        } else{
	    return stringReverse(s.substring(1), n-1) + s.charAt(0);
	}
    }
}

File: lab3/FileReverse.c
/*
 * Duy Nguyen
 * W1475318
 * CS 12M lab3
 * 10/17/16
 * FileReverse.c
 * Reads the input file and prints each segment in reverse on a separate line
 * in the output file
 */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void stringReverse(char* s){
    int i=0, temp, j=strlen(s)-1;
    for(i=0; i<j; i++){
	temp = s[i];
	s[i] = s[j];
	s[j] = temp;
	j--;
    }
}

int main(int argc, char* argv[]){
    FILE* in; /* file handle for input */
    FILE* out; /* file handle for output */
    char word[256]; /* char array to store words from the input file */

    /* check command line for corret number of arguments */
    if(argc != 3){
	printf("Usage: %s <input file> <output file>\n", argv[0]);
	exit(EXIT_FAILURE);
    }

    /* open input file for reading */
    in = fopen(argv[1], "r");
    if(in==NULL){
	printf("Unable to read from file %s\n", argv[1]);
	exit(EXIT_FAILURE);
    }

    /* open output file for writing */
    out = fopen(argv[2], "w");
    if(out==NULL){
	printf("Unable to write to file %s\n", argv[2]);
	exit(EXIT_FAILURE);
    }

    /* read words from input file, print on separate lines to output file */
    while(fscanf(in, " %s", word) != EOF){
	stringReverse(word);
	fprintf(out, "%s\n", word);
    }

    /* close input and output files */
    fclose(in);
    fclose(out);

    return(EXIT_SUCCESS);
}
